feflow/tests/test_protein_mutation.py

```diff
@@ -192,3 +363,124 @@ def test_asp_to_leu_execute(self, protocol_asp_to_leu_result):
         # the FinishUnit will always be the last to execute
         finishresult = dagresult.protocol_unit_results[-1]
         assert finishresult.name == "result"
+
+    @pytest.mark.slow
+    def test_ala_gly_convergence(
```

Comment:
This test shares a lot of the same code as https://github.com/OpenFreeEnergy/feflow/pull/95/files#diff-82d94bde95c44db8b1006ee5251daf53bea07e1026ed5a559da150c720b090deR263 should it be using the `execute_forward_reverse_dag` function?

Reply:
Yes it definitely should, good catch.

Reply:
Actually, on a second thought, it's slightly different because of how the fe estimates and errors are computed for each case. There are ways to decrement the redundant code here but that requires probably better data types/containers to be able to be reused in both cases which I don't think it's worth it for the tests. That is, I think the tests are more readable this way, even if they have some redundant code. But as it's not API it should be fine, imo.

Reply:
I just remember the perses tests got really annoying to work with since so many tests were duplicated with just one little change. 

I think if you changed `execute_forward_reverse_dag` to `return forward_dagresult, reverse_dagresult` instead of the sums then each test would only be the different ways to caclualte errors.

Reply:
This should have been addressed in the latest comments. Thanks for the feedback!